common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -288,10 +303,27 @@ bool base_wlan_hal_dwpal::fsm_setup()
                     LOG(ERROR) << "Can't get event fd since dwpal ctx is NULL!";
                     return (transition.change_destination(dwpal_fsm_state::Detach));
                 }
-
                 // Success
                 LOG(DEBUG)
                     << "Open and attach an event interface to wpa_supplicant/hostapd - SUCCESS!";
+
+                // Get the nl event interface file descriptor
+                if (!m_dwpal_nl_ctx) {
```

Comment:
The dwpal netlink context is created in the `Init` state. On failure, we transition the FSM to the `Detach` state.
In what scenario do we get to the `Attach` state without a valid context?
I think this `if` statement is redundant.

Reply:
not sure it this test is mandatory.
I added it since we do the same with the VAPs .

Reply:
Still, this should be removed as part of #665

Reply:
here is a live example why it should stay.

netgear-rax40-1_logs:
DEBUG 17:21:12:420 <1991208208> 
base_wlan_hal_dwpal.cpp[148] --> **dwpal_driver_nl_attach() success for radio =wlan0**
...
DEBUG 17:21:13:264 <1991208208> 
base_wlan_hal_dwpal.cpp[394] --> DWPAL FSM wlan0 Attach State: Attach
INFO 17:21:13:264 <1991208208> 
ap_manager_thread.cpp[343] --> waiting to attach to wlan0
DEBUG 17:21:13:665 <1991208208> 
base_wlan_hal_dwpal.cpp[306] --> Open and attach an event interface to wpa_supplicant/hostapd - SUCCESS!
ERROR 17:21:13:665 <1991208208> 
base_wlan_hal_dwpal.cpp[322] --> **Can't get event fd since dwpal ctx is NULL!**
INFO 17:21:13:666 <1991208208> 
beerocks_state_machine.h[53] --> 'Attach': Overriding destination for 'Attach' from 'Operational' to 'Detach'
DEBUG 17:21:13:666 <1991208208> 
base_wlan_hal_dwpal.cpp[348] --> dwpal_fsm_state::Detach

Reply:
Exactly. Remember that RAX40 has a much older driver version at the moment so the attach *will* fail
We mustn't fail the FSM on that! Just log an warning and proceed without NL. Similar to what we do when we lack permissions to open raw socket - we simply don't open it.